datashader/core.py

```diff
@@ -264,7 +264,11 @@ def raster(self,
         w = int(np.ceil(self.plot_width * width_ratio))
         h = int(np.ceil(self.plot_height * height_ratio))
 
-        data = resample_2d(source.values[0].astype(np.float32),
+        rmin, cmin = get_indices(xmin, ymin, source.x.values, source.y.values, res)
+        rmax, cmax = get_indices(xmax, ymax, source.x.values, source.y.values, res)
```

Comment:
By "this" I mean inverting rmin/cmin and rmax/cmax. Happy to push the fix to this PR, but would be good if @gbrener could confirm.